js/widgets/forms/clearButton.js

```diff
@@ -24,55 +27,75 @@ define( [
 				this.options.clearBtn = true;
 			}
 
-			if ( !!this.options.clearBtn && this.inputNeedsWrap ) {
-				this._addClearBtn();
+			// We do nothing on startup if the options is off or if this is not a wrapped input
+			if ( !this.options.clearBtn || !this.inputNeedsWrap ) {
+				return;
+			}
+
+			if ( this.options.enhanced ) {
+				this._clearButton = this._outer.children( "a.ui-textinput-clear-button" );
+				this._clearButtonIcon = this._clearButton
+					.children( "span.ui-textinput-clear-button-icon" );
+				this._toggleClasses( "add" );
+				this._bindClearEvents();
+			} else {
+				this._addClearButton();
 			}
 		},
 
 		clearButton: function() {
-			return $( "<a href='#' tabindex='-1' aria-hidden='true' " +
-				"class='ui-input-clear ui-button ui-icon-delete ui-button-icon-only ui-corner-all'>" +
-				"</a>" )
+			var icon = $( "<span>" ),
+				button = $( "<a href='#' tabindex='-1' aria-hidden='true'></a>" )
 					.attr( "title", this.options.clearBtnText )
-					.text( this.options.clearBtnText );
+					.text( this.options.clearBtnText )
+					.append( icon );
+
+			return {
+				_clearButton: button,
+				_clearButtonIcon: icon
+			};
 		},
 
-		_clearBtnClick: function( event ) {
+		_clearButtonClick: function( event ) {
 			this.element.val( "" )
 					.focus()
 					.trigger( "change" );
-
-			this._clearBtn.addClass( "ui-input-clear-hidden" );
 			event.preventDefault();
 		},
 
-		_addClearBtn: function() {
-
-			if ( !this.options.enhanced ) {
-				this._enhanceClear();
-			}
-
-			$.extend( this, {
-				_clearBtn: this.widget().find("a.ui-input-clear")
-			});
+		_toggleClasses: function( operation ) {
+			this[ "_" + operation + "Class" ]( this._outer, "ui-textinput-has-clear-button" );
+			this[ "_" + operation + "Class" ]( this._clearButton, "ui-textinput-clear-button",
+				"ui-button ui-button-icon-only ui-button-right" );
+			this[ "_" + operation + "Class" ]( this._clearButtonIcon,
+				"ui-textinput-clear-button-icon",
+				"ui-icon-delete ui-icon" );
+		},
 
+		_addClearButton: function() {
+			this._addClass( this._outer, "ui-textinput-has-clear-button" );
+			$.extend( this, this.clearButton() );
```

Comment:
This just seems odd to have this return an object with the button and icon. we never documented this extension point and it does not serve much purpose as an extension point, when it returns an empty link containing an empty span, and this method only has 3 other lines. Why dont we just inline this.